File: services/identificacao-api/2-Application/Identificacao.Application/Dashboard/DashboardResumoResponse.cs
namespace Identificacao.Application.Dashboard;

/// <summary>
/// Resposta do resumo da dashboard para o domínio de Identificação.
/// Contém métricas de match, pendentes e captações.
/// </summary>
public record DashboardResumoResponse(
    double TaxaMatch,
    long TotalPendentes,
    int CaptacoesAtivas,
    string? UltimoLoteDescricao,
    IReadOnlyList<DashboardAlerta> Alertas);

public record DashboardAlerta(string Tipo, string Mensagem);

File: services/distribuicao-api/distribuicao-application/src/main/java/br/com/ecad/distribuicao/application/dto/DashboardResumoResponse.java
package br.com.ecad.distribuicao.application.dto;

import java.math.BigDecimal;
import java.util.List;

/**
 * Resposta do resumo da dashboard para o domínio de Distribuição.
 * Retorna métricas do último ciclo de distribuição e contagem de rubricas.
 */
public record DashboardResumoResponse(
        String statusUltimoCiclo,
        BigDecimal totalRepassado,
        BigDecimal creditosRetidos,
        int rubricasAtivas,
        List<DashboardAlerta> alertas) {

    public record DashboardAlerta(String tipo, String mensagem) {}
}
